topology/testutil/topology.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package testutil
+
+import (
+	"fmt"
+
+	"github.com/m3db/m3cluster/shard"
+	"github.com/m3db/m3db/sharding"
+	"github.com/m3db/m3db/topology"
+)
+
+// MustNewTopologyMap returns a new topology.Map with provided parameters.
+// It's a utility method to make tests easier to write.
+func MustNewTopologyMap(
```

Comment:
This is mostly personal preference, but since its in a test package you could accept testing.T which would allow you to do require.NoError(t, err.Error()) or t.Fatal or whatever

Reply:
Yeah i thought about doing that but I prefer not taking the dep as much as possible. the `MustPattern` is pretty standard and achieves the same thing.